src/main/kotlin/no/nb/mlt/wls/domain/WLSService.kt

```diff
@@ -76,6 +78,67 @@ class WLSService(
         return movedItem
     }
 
+    override suspend fun pickItems(
+        hostName: HostName,
+        itemsPickedMap: Map<String, Double>
+    ) {
+        val itemIds =
+            itemsPickedMap.map {
+                ItemId(hostName, it.key)
+            }
+
+        if (!itemRepository.doesEveryItemExist(itemIds)) {
+            throw ItemNotFoundException("Some items do not exist in the database, and were unable to be picked")
+        }
+
+        itemIds.map { itemId ->
+            val item = getItem(itemId.hostName, itemId.hostId)!!
+            if (itemsPickedMap[item.hostId] == null) {
+                // TODO - Review this. This state should generally not happen with the preconditions. Does it need to be handled at all?
+                throw ItemNotFoundException("Item with ID ${itemId.hostId} for host ${item.hostName} was not found, despite existing!")
+            }
+
+            val itemsStocked = item.quantity ?: 0.0
+            val itemsPicked = itemsPickedMap[item.hostId] ?: 0.0
+
+            // In the case of over-picking, set quantity to zero
+            // so that on return the database hopefully recovers
+            if (itemsPicked > itemsStocked) {
+                logger.error {
+                    "Tried to pick too many items for ${item.hostId}. WLS DB has $itemsStocked stocked, and storage system tried to pick $itemsPicked"
+                }
+            }
+            val movedItem =
+                itemRepository.moveItem(
+                    item.hostId,
+                    item.hostName,
+                    Math.clamp(itemsPicked.minus(itemsStocked), 0.0, Double.MAX_VALUE),
```

Comment:
Shouldn't that be other way around: itemsStocked.minus(itemsPicked)?
Since move item function sets the new quantity and location, so it should be 'new quantity' = 'number of items in stock' - 'number of picked items'

Reply:
Correct